src/test/components/panels/EventOverviews.test.tsx

```diff
@@ -0,0 +1,217 @@
+  "Attendance",
+];
+
+const getEventDetails = () => {
+  const table = screen.getByRole("table");
+  const rows = screen.getAllByRole("row", table);
+  const eventRow = rows[1];
+  const tableCells = within(eventRow).getAllByRole("cell");
+  return { eventRow, tableCells };
+};
+
+const getExpectedValues = ({
+  event,
+  type,
+}: {
+  event: (typeof mockFutureEventOverviews.results)[0];
+  type: "Past" | "Future";
+}) => {
+  switch (type) {
+    case "Past":
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Tue, 12 Sept 2023, 16:30",
+        "Mon, 11 Sept 2023, 16:30",
+        event.location?.address?.town,
+        event.eventStatus,
+        `${event.numberOfConfirmedBookings} / ${event.numberOfPlaces}`,
+        event.numberOfWaitingListBookings,
+        event.numberAttended,
+        event.numberAbsent,
+        `${
+          (event.numberAttended / (event.numberAttended + event.numberAbsent)) *
+          100
+        }%`,
+      ];
+    case "Future":
+    default:
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Thu, 9 Sept 9999, 20:00",
+        "Thu, 9 Sept 9999, 20:00",
+        event.location?.address?.town,
+        event.eventStatus,
+        `${event.numberOfConfirmedBookings} / ${event.numberOfPlaces}`,
+        event.numberOfWaitingListBookings,
+        event.numberAttended,
+        event.numberAbsent,
+        "-",
+      ];
+  }
+};
+
+describe("Admin Event Manager", () => {
+  const mockSetSelectedEventId = jest.fn();
+
+  const setupTest = (role: TestUserRole) => {
+    renderTestEnvironment({
+      role: role,
+      PageComponent: EventOverviews,
+      componentProps: {
+        setSelectedEventId: mockSetSelectedEventId,
+        user: { ...mockUser, loggedIn: true, role: role },
+      },
+      initialRouteEntries: ["/admin/events"],
+      extraEndpoints: [
+        rest.get(API_PATH + "/events/overview", (req, res, ctx) => {
+          const filter = req.url.searchParams.get("filter");
+          if (filter === "FUTURE") {
+            return res(ctx.status(200), ctx.json(mockFutureEventOverviews));
+          } else if (filter === "PAST") {
+            return res(ctx.status(200), ctx.json(mockPastEventOverviews));
+          }
+        }),
+      ],
+    });
+  };
+
+  it("renders for Admin user without crashing", () => {
+    setupTest("ADMIN");
+    const subtitle = screen.getByText("Events overview");
+    expect(subtitle).toBeInTheDocument();
+  });
+
+  it("renders warning for Event Leader user", () => {
+    setupTest("EVENT_LEADER");
+    const eventLeaderWarning = screen.queryByText(
+      "As an event leader, you are only able to see the details of events which you manage."
+    );
+    expect(eventLeaderWarning).toBeInTheDocument();
+  });
+
+  it("displays table with correct headers", async () => {
+    setupTest("ADMIN");
+    await screen.findByText("Actions");
+    headers.forEach((header) => {
```

Comment:
Since the `headers` constant is only used here, it would be better to move it inside this test. Alternatively, you can transform this test in a parametrised test.

Reply:
Shuffled the `headers` constant into the test now. I considered doing it.each but I don't think I need to do a re-render for checking each header; seemed more efficient to do `.forEach` after a single render.